expo/tests/scheduler/test_edit_event.py

```diff
@@ -1,97 +1,329 @@
+        self.assertIn('<li>Select a valid choice. That choice is not one of ' +
+                      'the available choices.</li>',
+                      response.content)
+
+    def test_good_user_with_duration(self):
+        login_as(self.privileged_profile, self)
+        url = reverse(self.view_name,
+                      urlconf="scheduler.urls",
+                      args=["Class", self.context.sched_event.pk])
+        form_data = get_sched_event_form(self.context)
+        form_data['event-duration'] = "3:00:00"
+        response = self.client.post(
+            url,
+            data=form_data,
+            follow=True)
+
+        self.assert_good_post(response,
+                              self.context,
+                              self.context.days[0],
+                              self.context.room)
+        self.assertIn('<input id="id_event-duration" name="event-duration" ' +
+                      'type="text" value="03:00:00" />',
+                      response.content)
+
+    def test_good_user_with_teacher(self):
+        overcommitter = PersonaFactory()
+        login_as(self.privileged_profile, self)
+        url = reverse(self.view_name,
+                      urlconf="scheduler.urls",
+                      args=["Class", self.context.sched_event.pk])
+        form_data = get_sched_event_form(self.context)
+        form_data['event-teacher'] = overcommitter.pk
+        response = self.client.post(
+            url,
+            data=form_data,
+            follow=True)
+
+        self.assert_good_post(response,
+                              self.context,
+                              self.context.days[0],
+                              self.context.room)
+        teachers = self.context.sched_event.get_direct_workers('Teacher')
+        self.assertEqual(len(teachers), 1)
+        self.assertEqual(teachers[0].pk, overcommitter.pk)
+        self.assertIn('<option value="' + str(overcommitter.pk) +
+                      '" selected="selected">' + str(overcommitter) +
+                      '</option>',
+                      response.content)
+
+    def test_good_user_remove_teacher(self):
+        login_as(self.privileged_profile, self)
+        url = reverse(self.view_name,
+                      urlconf="scheduler.urls",
+                      args=["Class", self.context.sched_event.pk])
+        form_data = get_sched_event_form(self.context)
+        form_data['event-teacher'] = ""
+        response = self.client.post(
+            url,
+            data=form_data,
+            follow=True)
+
+        self.assert_good_post(response,
+                              self.context,
+                              self.context.days[0],
+                              self.context.room)
+        teachers = self.context.sched_event.get_direct_workers('Teacher')
+        self.assertEqual(len(teachers), 0)
+        self.assertIn('<select id="id_event-teacher" name="event-teacher">\n' +
```

Comment:
FWIW, python concatenates adjacent strings, so the + here is not necessary. Not a blocker, but worth knowing.